c/include/arrow-adbc/adbc.h

```diff
@@ -152,15 +152,19 @@ struct ArrowArrayStream {
 // Storage class macros for Windows
 // Allow overriding/aliasing with application-defined macros
 #if !defined(ADBC_EXPORT)
-#if defined(_WIN32)
+#if (defined(_WIN32) || defined(__CYGWIN__)) && defined(ADBC_BUILD_DLL)
```

Comment:
While this is consistent with nanoarrow, I do wonder if requiring two defines (ADBC_EXPORTING and ADBC_BUILD_DLL) is superfluous; the latter prevents the former from having any meaning when applied to a static library, but maybe it would be easier to just ensure ADBC_EXPORTING is never defined for static targets (?)

Reply:
`ADBC_BUILD_DLL` is needed because we can't use `__declspec(dll{import,export})` entirely for static linking.

FYI: Apache Arrow C++ uses `ARROW_STATIC` not `ADBC_BUILD_DLL`: https://github.com/apache/arrow/blob/6e84d990379a0fe20a0a89311aa864f40efded23/cpp/src/arrow/util/visibility.h#L42

It's specified when static linking not shared linking.

I think that specifying additional macro for building static linking (`ARROW_STATIC`) approach is easier to use than specifying additional macro for building shared library (`ADBC_BUILD_DLL`). Because we can add additional cflags with `.pc` by `Cflags.private`.

Reply:
Is the idea that downstream targets that link to a static arrow library would also need to define `ADBC_STATIC`? The current design necessitates that on Windows, but I'm not sure that is correct